### ISSUE TITLE
`coqc --config` should not fail when the stdlib cannot be found

### ISSUE THREAD

**Issue by JasonGross** [coq/coq#16295]
```
$ coqc --config
  cannot guess a path for Coq libraries; please use -coqlib option or ensure you have installed the package containing Coq's stdlib (coq-stdlib in OPAM) If you intend to use Coq without a standard library, the -boot -noinit options must be used.
```
This makes it really hard to debug what's going wrong

**Comment by Alizter** [comment]
As a minimal, you will need to pass `-coqlib` that points to a directory containing `theories/Init/Prelude.v` (or `vo`?).

cc @ejgallego

**Comment by Alizter** [comment]
@JasonGross Can you provide some more information about how you ran into this? AFAICT this only happens if `./configure` never gets run or `theories/Init/Prelude.vo` cannot be found. Both of which are difficult to do in practice. I would be interested to know about your use case.

**Comment by Alizter** [comment]
@ejgallego Could you open an issue detailing your ideas then? I also had a look at this today, but I concluded that the issue stems from more fundamental design flaws with how coqlib is detected, passed around etc. It is not an easy fix.

**Comment by ejgallego** [comment]
Yes, it is not an easy fix. Not worth opening an issue IMO, some ideas are:
- forbid calling `Env.init` more than once per execution, once you get the env, pass it around to users.
- reflect failure in types, and callers handle the errors
- phase split, in the sense of `Env.config` returning the values Coq understands by default, however, that would mean a few functions having to fail later on, for example, when asking for `Env.tool`.

So indeed, maybe we could just remove the "I cannot find stlib" error at all, and just have errors happen for example for prelude at import time: "I cannot find `foo/Init/Prelude.vo`.

**Comment by Alizter** [comment]
>forbid calling Env.init more than once per execution, once you get the env, pass it around to users.

This is in effect, how it works today. If you call Env.Init after it has already been initialized, it detects the reference being set and just passes the already set one.

>reflect failure in types, and callers handle the errors

This should be doable.

>So indeed, maybe we could just remove the "I cannot find stlib" error at all, and just have errors happen for example for prelude at import time: "I cannot find foo/Init/Prelude.vo.

Sounds good.

**Comment by ejgallego** [comment]
That's good, but indeed we build stdlib with -boot (which is actually our goal medium term), so it is normal Env.init is never called.

I'm more curious about regular developments, for example, doing a `git grep Env.init` I can see a few extra calls that we'd have to refactor:
```
kernel/nativelib.ml:68:                  let env = Boot.Env.init () in
lib/envars.ml:171:                       let env = Boot.Env.init () in
plugins/micromega/coq_micromega.ml:2189: let env = Boot.Env.init () in
sysinit/coqargs.ml:449:                  let coqenv = Boot.Env.init () in
sysinit/coqinit.ml:92:                   let env = Boot.Env.init () in
toplevel/coqloop.ml:517:                 let env = Boot.Env.init () in
toplevel/coqtop.ml:22:                   let env = Boot.Env.init () in

```
Once we have that, we can make `Env.init` take some arguments and do less guessing.

That would solve this bug actually as for example the one in `lib/envars` would go away.
